jax/_src/scipy/special.py

```diff
@@ -2185,19 +2215,125 @@ def ci_series(x):
     t = x * x
     return np.euler_gamma + jnp.log(x) + t * jnp.polyval(CN, t) / jnp.polyval(CD, t)
 
-  si = jnp.piecewise(
-    jnp.abs(x),
-    [x == 0, jnp.isinf(x)],
```

Comment:
Similar to above comments, removing this inf check because it's not within the accepted range.